### DISCUSSION TITLE
Output pasture production

### DISCUSSION THREAD

**Discussion by KeLiu7** [magpiemodel/magpie#447] [Help] [answered]
Dear MAgPIE community,

I'm trying to extract pasture yield and plot it on a global map with Panoply. Could you please help me check if my scripts below are right? 

Also, could you please tell me which variable I should write for `weight =?` when using `gdxAggregate` for disaggregating? I'm not sure if I should write pasture here.

Many thanks!

past_pro<- collapseNames(readGDX(gdx="fulldata.gdx","oq14_yield_past")[,,"level"])

grid_pastproduc <- gdxAggregate(gdx="fulldata.gdx",x = past_pro, to="grid",weight=pasture, absolute=TRUE,dir=".")

write.magpie(grid_pastproduc,"grid_pastproduc.nc")

**Comment by abhimishr** [comment, marked as answer]
Reading `oq14_yield_past` is not the right way to read pasture yields as you are reading the values of the equation. 
What you need in this case is `vm_yld` variable for `pasture`.
This should be read in as `ov_yld` and then you can filter for `pasture`. 

Here, the weight argument has to be a 0.5 degree spatial scale magpie object most likely. Other MAgPIE members might have a good suggestion for that. Note that pasture yields are relative measures, so the choice of weight is critical in interpreting the yield results on a 0.5 degree scale.

**Reply by KeLiu7** [reply]
Many thanks Abhimishr for your suggestion! I will have another try and keep that in mind.
